### ISSUE TITLE
Bug in the workflow execute.py module

### ISSUE THREAD

**Issue by wjcbsr** [Commvault/ansiblev2#8]
There seems to be an issue with the job_id variable in execute.py

When I run my defined workflow using ansiblev2 the workflow is executed properly on the back-end but my playbook fails with the following error:

TASK [ebs-nfs-backup : Backup PAP2 NFS application filesystems] ********************************************************************************************************************************************
fatal: [rlx7200.corp.ads]: FAILED! => {"changed": false, "msg": "**'tuple' object has no attribute 'job_id'**"}

This is my simple task:
- name: Backup PAP2 NFS application filesystems
  commvault.ansible.workflow.execute:
    workflow_name: "PAP2_NFS_backup"
    workflow_inputs:
    register: backup_job

**Comment by kevinjojeejoseph** [comment]
Hi @wjcbsr ,
Looks like the case where a job ID will not be returned after workflow execute is not handled. I will fix this

**Comment by wjcbsr** [comment]
It looks like the value is in there but not in the way it can be used:

"msg": "Value of unknown type: <class 'cvpysdk.job.Job'>, Job class
instance for job id: \"7802117\""

**Comment by kevinjojeejoseph** [comment]
Where do you see this message? I thought the message was "'tuple' object has no attribute 'job_id'" because the the return value was not unpacked correctly from the SDK call
https://github.com/Commvault/cvpysdk/blob/master/cvpysdk/workflow.py#L1254

**Comment by wjcbsr** [comment]
I just dumped the value of element [1] of the workflow_job to show it in my ansible output:
module.result['blah'] = workflow_job[1]

**Comment by kevinjojeejoseph** [comment]
Ok got it. This should be taken care of once unpacking is done correctly. Will fix this

**Comment by wjcbsr** [comment]
It looks like that is the job Id of the qlogin step in my workflow.  It completes almost instantly on the CV side.

**Comment by kevinjojeejoseph** [comment]
Should be fixed now with the latest commit

**Comment by wjcbsr** [comment]
Nope :-(  Same error:

"msg": "'tuple' object has no attribute 'job_id'"

Line 148 in execute.py fails.

**Comment by wjcbsr** [comment]
My workflow is very simple.  Just one QCommand->Operation->Execute step.  I ship the inputXML via ansible as an input.

The workflow/QCommand does get properly executed on the backend but ansible throws the error same as before.
![PAP2_NFS_backup](https://user-images.githubusercontent.com/37458321/216652701-18b661e9-0078-4354-a5c8-8f6a5ddbd360.png)

**Comment by kevinjojeejoseph** [comment]
Hmm, I'm not sure anymore. If you see the changes I've made on line 148, it should only enter the if block if workflow_job is an instance of cvpysdk.job.Job. Why is it saying it is a tuple? Something seems wrong

**Comment by wjcbsr** [comment]
My mistake Kevin.  Looks like I didn't have the fixes in where I needed to
put them.  It does succeed now.  The problem now is it looks like I don't
get anything back in the registered ansible variable from that
workflow.execute module :-(

**Comment by kevinjojeejoseph** [comment]
Can you please reach out to support@commvault.com?
We can get this checked with the staged CS that will provide an identical setup to yours so we can check what is going wrong and also help get the required logs

**Comment by wjcbsr** [comment]
You bet I can.  Thanks for all your help!

**Comment by wjcbsr** [comment]
It looks like the only agents I can backup and restore are “File System” agents.  This seems to be head-coded” in specific modules.

It would be great if we could have the option to backup “Linux File System” agents as well.

Can we add the agent as an argument to the backup module in ansiblev2 please?

Bill Beresnak
Stantec Consulting

**Comment by wjcbsr** [comment]
Hi William,

Thank you for contacting Commvault Customer Support. Unfortunately, Commvault Customer Support no longer supports the processing of cases/tickets through e-mail. 

This is a decision that has been handed down from upper management. The reason for stopping this service is since we are not able to collect enough information from the customer to fully create a case. Additionally due to the number of emails received daily, the response time is slower than the traditional methods of case creation. Below are the suggested methods of case creation. 

To ensure you receive the right priority and to verify that we have the right information to provide you the best possible experience on your support incident please call our Frontline team or create an incident directly via Maintenance Advantage.

• For a list of local support numbers, please click the link below:               
                ○ https://ma.commvault.com/Support/TelephoneSupport          
 • To open an incident directly on Maintenance Advantage, please click the link below:           
                ○ https://ma.commvault.com/Case/New               
• To discuss a general question or open a low priority issue, open a Chat with Support session via Maintenance Advantage:             
                ○ https://ma.commvault.com/ 

If you already have an open incident and need to adjust your priority or update your case, please use the above two methods. Have a great day! 

Mohamed Farhan 
Customer Support Associate, Frontline.





***@***.***



1800-419-2951
 

040-6654-0300


















-----Original Message-----
From: William B. ***@***.***> 
Sent: 09 February 2023 05:20
To: Support-Commvault ***@***.***>
Cc: Commvault/ansiblev2 ***@***.***>
Subject: CommVault/ansiblev2 Enhancement Request

External email. Inspect before opening.



It looks like the only agents I can backup and restore are “File System” agents.  This seems to be head-coded” in specific modules.

It would be great if we could have the option to backup “Linux File System” agents as well.

Can we add the agent as an argument to the backup module in ansiblev2 please?

Bill Beresnak
Stantec Consulting
